### ISSUE TITLE
Pages without focus

### ISSUE THREAD

**Issue by jeffg2k** [jeffg2k/SmartCopy#35]
There are certain pages where it is difficult to determine who the focus is on the profile.  For example, MyHeritage marriage records are just a record of relationships, but you don't know if the focus is the bride, groom, or other.  Another is openarch, which presents a relationship tree without focus.

I'd like to have the ability to prompt the user and say something like "Who is the focus person in this page?" and give them a dropdown list of names that appear on the page.  Once a name is selected, the page can be parsed.

https://www.openarch.nl/show.php?archive=bhi&identifier=e4fd9a1b-2b8c-0ce1-e4aa-7854b61cd9ab&lang=en

https://www.myheritage.com/research/record-30010-957984/john-smith-and-margaret-a-huston-in-indiana-marriages?s=77335302

**Comment by stbodie** [comment]
I can't recall what SC currently does on those pages - I think it just says it's unsupported, so probably wouldn't be too hard to include a prompt or try to auto-detect based on names, sex, etc.

**Comment by DanCornett** [comment]
Yes, currently it says the page is unsupported. MH marriage pages are easiest examples to find.

**Comment by stbodie** [comment]
> Yes, currently it says the page is unsupported. MH marriage pages are easiest examples to find.

@DanCornett do you have an example URL for me to look into?  I tested a marriage record on the MyHeritage results page that GENI spawns and it worked for me... are you talking about a page directly on MyHeritage.   

AND, could this also apply to birth/marriage/death records on Ancestry?

**Comment by stbodie** [comment]
Confirmed - same root cause as this issue's own MyHeritage example (`record-30010-957984/...`), just a different specific record: `research/record-21028-2162160-/leo-hamlisch-and-henrietta-flesdrager-in-new-york-city-marriages`.

**Why the green checkmark shows but the page still fails:** `myheritage.com` is unconditionally in `background.js`'s `SUPPORTED_SITE_HOSTS` list, so the toolbar badge only reflects "this domain has *some* MyHeritage support," not "this specific page can be parsed." Neither registered MyHeritage collection recognizes a `research/record-` URL:
- `collections/myheritage.js`'s `collectionMatch()` only matches `person-`/`member-`/`site-family-tree-` (the legacy profile URL shapes).
- `collections/myheritagenew.js`'s `collectionMatch()` only matches `profile-` (the redesigned profile page).

A `/research/record-.../...-marriages` URL is neither - it's a search-index/record-transcription page describing a historical document (a marriage record naming a bride and groom), not a person profile page at all. So it correctly falls through to the generic "not yet supported" message once neither `collectionMatch()` fires - that part of the flow is working as designed, just not helpfully in this case.

**What it would take to actually parse this page:** two separate pieces of work, not one:
1. A new (or extended) collection with a `collectionMatch()` recognizing `research/record-` and a parser reading the record page's DOM - marriage date, marriage location, and the two names, at minimum.
2. The actual subject of this issue: MyHeritage marriage records have **two** people (bride + groom) and no inherent "this is the focus profile" concept the way a person-profile page does. Parsing the record alone doesn't resolve who SmartCopy should treat as the focus - that's exactly the "Who is the focus person on this page?" prompt/dropdown this issue is asking for. Without that piece, a record parser would have nowhere to point the parsed data.

So this isn't a second bug alongside the original report - it's confirmation the underlying gap (record-type pages with no single focus) shows up on more URL shapes than just the one originally reported, and a live example of exactly the "easiest examples to find" @DanCornett flagged. Fixing it means implementing this issue's own proposal (the focus-picker prompt), not a smaller independent patch.

**Comment by stbodie** [comment]
**Correction to my last comment** - I mis-diagnosed the root cause. `collections/smartmatch.js` *does* register a `collectionMatch()` for `research/record-` URLs (I'd missed that this codebase has a third MyHeritage collection file, separate from `myheritage.js`/`myheritagenew.js`). My apologies for the noise.

The real cause: `parseSmartMatch()` (the parser that collection uses) has had this at the very top since the file's original 2014 commit -
```js
if ($(htmlstring).filter('title').text().contains("Marriages")) {
    setMessage(warningmsg, 'This MyHeritage collection is not yet supported by SmartCopy.');
    return "";
}
```
It unconditionally refuses *any* record whose page title contains "Marriages" - almost certainly for exactly the reason this issue exists (bride/groom, no inherent focus).

Here's the part that made it worth fixing outright rather than leaving alone: this guard fires **even when a focus has already been resolved**. `parseSmartMatch`/`parseProfileData` is only ever called from `popup.js`'s `loadPage()`, and only from the branch that already requires a confirmed `focusid` - reached via a MyHeritage "In my Geni tree" sidebar match (`.individualInformationProfileLink` - this file already reads that exact widget elsewhere, in its own `loadPage()`, for the non-marriage case), a prior SmartCopy submission history match, the tab's opener chain, or the user manually typing a Geni ID into "Set Destination". By the time this guard runs, one of those has *already* succeeded - the guard was throwing that away unconditionally, every time, regardless of how confidently focus was already known.

Concrete example that surfaced this: `https://www.myheritage.com/research/record-21028-2162160-/leo-hamlisch-and-henrietta-flesdrager-in-new-york-city-marriages` - the sidebar shows a MyHeritage Smart Match directly to `geni.com/profile-24661835` (Leo Hamlisch), so SmartCopy already knew the exact destination, but still refused the page outright.

**Fix pushed:** removed the unconditional refusal. When a focus is already known (any of the four ways above), the record now parses normally through the same generic field logic every other MyHeritage record already uses - no picker needed, since we already know. When nothing resolves a focus, it now falls through to the same generic "Set Geni Destination Profile" manual picker every other unresolved collection already uses, instead of a MyHeritage-specific dead end.

**Still open / unverified:** whether this record type's own field labels (marriage date/place, and however MyHeritage represents the *other* spouse, if at all as a separate field rather than just page-title text) get picked up as cleanly as birth/death records do - that needs a live test against a real marriage record, which I can't do from source alone. If anyone hits a record where the focus resolves correctly but a field comes through empty or attached to the wrong person, please paste the URL here.

This doesn't fully close the issue - the openarch case (and any MyHeritage/Ancestry record with *no* existing Geni match for either person) still has no focus-resolution path and needs the "who is this?" picker the issue originally proposed. This fix only covers the case where MyHeritage (or SmartCopy's own history/opener-chain/manual entry) already knows the answer.

**Comment by stbodie** [comment]
Closing this out. What ended up shipping is better than the original "who is the focus?" picker proposal: auto-detection, no picker needed.

For MyHeritage marriage records specifically, focus is now resolved automatically by name-matching against whichever Geni profile is already known (via the "In my Geni tree" sidebar match, submission history, opener chain, or a manually-set destination) - including a married-name fallback so setting a bride's own Geni profile directly still resolves correctly even though her marriage record names her under her maiden surname. When neither side of the record matches, it falls through to the same generic "Set Geni Destination Profile" picker every other unresolved collection already uses - so the picker this issue asked for effectively already exists as a fallback, just not as the primary mechanism.

The openarch.nl example from the original report is tracked separately under #94 (site support isn't built yet) - whoever implements that collection will need to solve focus-resolution for its own page structure, but that's out of scope here.

Summary of what shipped, in case useful for #94 or similar record-type work:
- `collections/smartmatch.js` - parses MyHeritage's Groom/Bride nested-detail table (a different shape from the flat one-field-per-row records this file otherwise handles)
- Focus side resolved by exact name match first, falling back to first-name + spouse's-surname matching (handles the maiden/married-name case) - deliberately never resolves via gender alone, since a same-gender third party (e.g. a parent mentioned only as a sub-field) could otherwise be misattributed
- When focus can't be resolved at all, falls through to the existing generic manual picker (`loadSelectPage()`) rather than a dead-end "not supported" message
